bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -1044,12 +1084,27 @@ private boolean setWebpageData(String url, String postData, String[] headers, St
 		if (stream != null) stream.Release();
 		if (hr != COM.S_OK) error(SWT.ERROR_NO_HANDLES, hr);
 		IUnknown request = new IUnknown(ppRequest[0]);
-		hr = webView_2.NavigateWithWebResourceRequest(request);
-		request.Release();
+		lastWebView_2Task = scheduleWebViewTask(lastWebView_2Task, wv -> {
+			wv.NavigateWithWebResourceRequest(request);
+			request.Release();
+		});
 	} else {
-		hr = webView.Navigate(pszUrl);
+		lastWebViewTask = scheduleWebViewTask(lastWebViewTask, wv -> wv.Navigate(pszUrl));
 	}
-	return hr == COM.S_OK;
+	return true;
+}
+
+/*
+ * Schedule a given consumer in the given webViewTask.
+ * Note: This will be executed in the thread from which
+ * it is called and must be therefore called from the UI thread.
+ */
+private <T extends ICoreWebView2> CompletableFuture<T> scheduleWebViewTask(CompletableFuture<T> webViewTask, Consumer<T> action) {
+	return webViewTask.thenApply(wv -> {
+		action.accept(wv);
+		browserResize(new Event());
```

Comment:
Why is a resize call triggered here? This is not obvious from the semantics indicated by the method signature and its documentation. It rather looks like a hack to achieve something undocumented.